src/devices/OpenHardwareMonitor/OpenHardwareMonitor.cs

```diff
@@ -0,0 +1,740 @@
+            }
+        }
+
+        /// <summary>
+        /// Stops all monitoring.
+        /// </summary>
+        public void StopAllMonitoring()
+        {
+            lock (_lock)
+            {
+                _monitoredElements.Clear();
+            }
+
+            if (_monitorThread != null)
+            {
+                _monitorThread.Join();
+                _monitorThread = null;
+            }
+        }
+
+        private void MonitorThread()
+        {
+            // Stops the thread when the list of monitored elements becomes empty
+            bool running = true;
+            while (running)
+            {
+                lock (_lock)
+                {
+                    DateTimeOffset now = DateTimeOffset.UtcNow;
+                    if (_lastMonitorLoop < now - TimeSpan.FromMinutes(5))
+                    {
+                        // We were apparently in sleep mode - skip this loop, because it will falsify results
+                        // (i.e. integrating power usage over a time where the computer was off will be incorrect)
+                        foreach (var elem in _monitoredElements)
+                        {
+                            elem.LastUpdated = now;
+                        }
+
+                        _lastMonitorLoop = now;
+                        continue;
+                    }
+
+                    foreach (var elem in _monitoredElements)
+                    {
+                        TimeSpan timeSinceLastUpdate = now - elem.LastUpdated;
+                        if (timeSinceLastUpdate > elem.Interval)
+                        {
+                            if (elem.Sensor.TryGetValue(out IQuantity value))
+                            {
+                                elem.OnNewValue(elem.Sensor, value, timeSinceLastUpdate);
+                            }
+
+                            elem.LastUpdated = now;
+                        }
+                    }
+
+                    _lastMonitorLoop = now;
+                    running = _monitoredElements.Count > 0;
+                }
+
+                if (running)
+                {
+                    Thread.Sleep(MonitorInterval);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Adds some special derived sensors.
+        /// - For each power sensor, this adds another sensor that integrates power over time and so generated the energy used in W/h or
+        /// more conveniently, Kilowatthours (this is the unit the electricity bill bases on)
+        /// - Gives the heat flux for the primary CPU, using the given CPU die size (or a default value)
+        /// </summary>
+        /// <param name="cpuDieSize">Die size of your CPU, optional. Find your CPU on https://en.wikichip.org/ to find out. Note: This
+        /// value is usually much smaller than the size of the physical CPU.</param>
+        public void EnableDerivedSensors(Area cpuDieSize = default)
```

Comment:
I really love this concept showing the power of having UnitNet and being able to make advance scenarios!